include/SDL3_shadercross/SDL_shadercross.h

```diff
@@ -64,16 +64,42 @@ typedef struct SDL_ShaderCross_ComputePipelineInfo
     Uint32 threadCountX;                 /**< The number of threads in the X dimension. */
     Uint32 threadCountY;                 /**< The number of threads in the Y dimension. */
     Uint32 threadCountZ;                 /**< The number of threads in the Z dimension. */
-} SDL_ShaderCross_ComputePipelineInfo;
+} SDL_ShaderCross_ComputePipelineMetadata;
+
+typedef struct SDL_ShaderCross_SPIRV_Info
+{
+    const Uint8 *bytecode;                    /**< The SPIRV bytecode. */
+    size_t bytecodeSize;                      /**< The length of the SPIRV bytecode. */
+    const char *entrypoint;                   /**< The entry point function name for the shader in UTF-8. */
+    SDL_ShaderCross_ShaderStage shaderStage;  /**< The shader stage to transpile the shader with. */
+    bool enableDebug;                         /**< Allows debug info to be emitted when relevant. Can be useful for graphics debuggers like RenderDoc. */
+    const char *name;                         /**< A UTF-8 name to associate with the shader. Optional, can be NULL. */
+
+    SDL_PropertiesID props;                   /**< A properties ID for extensions. Should be 0 if no extensions are needed. */
```

Comment:
What kind of props do you think will be needed in the future? I know, hard question. If you'd know, you'd have added them already.

Reply:
I think the main use of properties here would be if the shader compilers add functionality or if there's existing functionality we haven't covered.